Source/SDK/ClientBuilder.cs

```diff
@@ -364,5 +448,27 @@ async Task EventHorizonRetryPolicy(Subscription subscription, ILogger logger, Fu
                 await Task.Delay(timeout).ConfigureAwait(false);
             }
         }
+
+        ClientBuilder WithAllScannedAssemblies(Action<Assembly> doWithAssembly, Func<Assembly, bool> assemblyFilter = default, bool includeExeFiles = false)
+        {
+            foreach (var assembly in GetAssemblies(assemblyFilter, includeExeFiles))
+            {
+                doWithAssembly(assembly);
+            }
+
+            return this;
+        }
+
+        Task RegisterEventTypes(MethodCaller methodCaller, ExecutionContext executionContext, EventTypes eventTypes)
+            => new Events.Internal.EventTypes(methodCaller, executionContext, _loggerFactory.CreateLogger<Events.Internal.EventTypes>()).Register(eventTypes, _cancellation);
+
+        IEnumerable<Assembly> GetAssemblies(Func<Assembly, bool> assemblyFilter, bool includeExeFiles)
+        {
+            var logger = _loggerFactory.CreateLogger<ClientBuilder>();
+            return AssemblyFinder.FindAssemblies(
+                failedFile => logger.LogWarning("Failed to load assembly from file {File}", failedFile),
+                assemblyFilter ?? (_ => true),
+                includeExeFiles);
+        }
```

Comment:
A few Assembly loading things ended up in here now - maybe we should move them out to another thing?